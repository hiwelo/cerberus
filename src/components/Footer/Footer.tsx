import React, { FunctionComponent } from 'react';
import { StyledLink } from '..';
import { FooterContainer } from './components';

/**
 * Provides a footer giving information about the current application
 */
export const Footer: FunctionComponent = () => {
  return (
    <FooterContainer>
      Provided by{' '}
      <StyledLink href="https://github.com/hiwelo/cerberus">
        Cerberus
      </StyledLink>
      , made with love by{' '}
      <StyledLink href="https://github.com/hiwelo">hiwelo</StyledLink> in
      Rotterdam.
    </FooterContainer>
  );
};
