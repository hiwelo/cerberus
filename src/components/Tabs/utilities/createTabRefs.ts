import { useRef } from 'react';
import { TabsProps } from '../types';

/**
 * Creates a series of React Ref for each panel part of the Tabs defined for
 * this `Tabs` component
 *
 * @param tabs The dictionary of Tabs for the related Tabs component
 */
export function createTabRefs(tabs: TabsProps['tabs']): TabsProps['tabs'] {
  // Transforms the TabListDictionary as an array of tabs
  const tabsEntries = Object.entries(tabs);
  const tabsWithRefs: TabsProps['tabs'] = {};

  // Loops over each tab and creates a React Ref for each panel HTMLElement
  tabsEntries.forEach(([tabKey, tabInfo]) => {
    const refPanel = useRef<HTMLDivElement>(null);

    tabsWithRefs[tabKey] = { ...tabInfo, refPanel };
  });

  return tabsWithRefs;
}
