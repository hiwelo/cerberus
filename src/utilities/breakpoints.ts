/** List of breakpoints in use in our styled components */
export const breakpoints = {
  xs: '0px',
  md: '540px',
};
