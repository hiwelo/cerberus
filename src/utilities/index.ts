export { breakpoints } from './breakpoints';
export { listNodesFromConnection } from './listNodesFromConnection';
