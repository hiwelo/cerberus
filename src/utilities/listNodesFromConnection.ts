/** Describes the generic structure of a GraphQL edge */
type GenericEdge<T> = {
  /** Item from a GraphQL connection object */
  node: T;
  /** Potential typing for the edge added by GraphQL */
  __typename: string;
};

/** Describes the generic structure of a GraphQL connection object */
type GenericConnection<T> = {
  edges: GenericEdge<T>[];
};

/**
 * Returns an array of all nodes contained in the provided GraphQL connection
 * object
 *
 * @param connection GraphQL connection object containing the nodes to return
 */
export function listNodesFromConnection<T>(
  connection?: GenericConnection<T>,
): T[] | undefined {
  return connection?.edges.map(({ node }) => node);
}
