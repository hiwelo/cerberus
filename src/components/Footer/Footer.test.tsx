import React from 'react';
import { testA11y, testSnapshot } from 'utilities/test';
import { Footer } from './Footer';

describe('Footer', () => {
  testSnapshot(<Footer />);
  testA11y(<Footer />);
});
