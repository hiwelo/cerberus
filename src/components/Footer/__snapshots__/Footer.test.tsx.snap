// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
.c1 {
  position: relative;
  color: #006e97;
  font-weight: 500;
}

.c1:focus {
  outline: none;
}

.c3 {
  position: relative;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3::before {
  position: absolute;
  bottom: 0.1em;
  left: 0;
  right: 0;
  content: '';
  height: 0.15em;
  background-color: #e8c000;
  border-radius: 0.4rem;
}

.c3:focus {
  outline: none;
}

.c3:focus::before,
.c3:hover::before {
  background-color: #8ab0ab;
  bottom: 0;
}

.c0 {
  margin: 1.6rem auto 0.8rem;
  padding: 0 1.6rem;
  font-size: 1rem;
}

.c0 .c2 {
  color: #222222;
}

.c0 .c2::before {
  background-color: currentColor;
  height: 0.1em;
}

@media (prefers-reduced-motion:no-preference) {
  .c3::before {
    -webkit-transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
    transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
  }
}

<footer
  class="c0"
>
  Provided by 
  <a
    class="c1 c2 c3"
    href="https://github.com/hiwelo/cerberus"
  >
    Cerberus
  </a>
  , made with love by 
  <a
    class="c1 c2 c3"
    href="https://github.com/hiwelo"
  >
    hiwelo
  </a>
   in Rotterdam.
</footer>
`;
